core/types/block.go

```diff
@@ -90,6 +90,9 @@ type Header struct {
 
 	// DataGasUsed was added by EIP-4844 and is ignored in legacy headers.
 	DataGasUsed *uint64 `json:"dataGasUsed" rlp:"optional"`
+
+	// BeaconRoot was added by EIP-4788 and is ignored in legacy headers.
+	BeaconRoot *common.Hash `json:"beaconRoot" rlp:"optional"`
```

Comment:
Adding it like this delegates the responsibility of "when to start storing data for EIP-4788" entirely to the CL layer. When/of they start including a beaconroot, we start storing it into the state. 

IMO we should 
- only accept BeaconRoot after the fork time, 
- require the BeaconRoot after the fork time, 

As for RLP, similarly, we need to be careful, and strict.

Reply:
Is the order between BeaconRoot and DataGasUsed specified? Both are activated in the same fork...

Reply:
I don't think its specified yet

Reply:
it should be at the end of all the 4844 stuff; I was trying to specify this abstractly to avoid thrash from other EIPs but I can see how this is unclear so I made it explicit:

https://github.com/ethereum/EIPs/pull/7297